src/main/java/com/butbetter/storage/csvImport/controller/FileUploadController.java

```diff
@@ -92,11 +80,50 @@ public FileUploadController(IFileStorageService fileStorageService) {
 	 *
 	 * @param e FaultyCSVException
 	 *
-	 * @return Build Response-Entity
+	 * @return build Response-Entity with bad request and information of stacktrace
 	 */
 	@ExceptionHandler(FaultyCSVException.class)
 	public ResponseEntity<?> handleFaultyCSVException(FaultyCSVException e) {
 		logger.error(e.getMessage());
 		return ResponseEntity.badRequest().body(e.getMessage());
 	}
+
+	/**
+	 * Controller Exception Handler for CsvException
+	 *
+	 * @param e CsvException
+
+	 * @return build Response-Entity with bad request and information of stacktrace
+	 */
+	@ExceptionHandler(CsvException.class)
+	public ResponseEntity<?> handleGeneralCsvException(CsvException e) {
+		logger.error(e.getMessage());
+		return ResponseEntity.badRequest().body(e.getMessage());
+	}
+
+	/**
+	 * Controller Exception Handler for StorageException
+	 *
+	 * @param e StorageException
+
+	 * @return build Response-Entity with bad request and information of stacktrace
+	 */
+	@ExceptionHandler(StorageException.class)
+	public ResponseEntity<?> handleStorageException(StorageException e) {
+		logger.error(e.getMessage());
+		return ResponseEntity.badRequest().body(e.getMessage());
+	}
+
+	/**
+	 * Controller Exception Handler for StorageFileNotFoundException
+	 *
+	 * @param e StorageFileNotFoundException
+
+	 * @return build Response-Entity with internal server error and information of stacktrace
+	 */
+	@ExceptionHandler(StorageFileNotFoundException.class)
+	public ResponseEntity<?> handleStorageFileNotFoundException(StorageFileNotFoundException e) {
+		logger.error(e.getMessage());
+		return ResponseEntity.internalServerError().body(e.getMessage());
```

Comment:
I think best choice is bad request (400) because the internal server error is an unexpected condition.
Link: https://datatracker.ietf.org/doc/html/rfc7231#section-6.6.1

Reply:
Oh yeah, that's on me, 
the naming might be a bit off, will define another exception for internal server error and add it where it's needed